File: groupmates.py
#coding:utf-8

groupmates = [
    {
        "name" : u"Гасан",
        "group" : u"БСТ1703",
        "age" : 18,
        "marks" : [4, 4, 3, 5]
    },

    {
        "name" : u"Марьяна",
        "group" : u"БСТ1703",
        "age" : 20,
        "marks" : [2, 5, 4, 5]
    },

    {
        "name" : u"Кирилл",
        "group" : u"БСТ1703",
        "age" : 19,
        "marks": [3, 4, 4, 4]
    },

    {
        "name" : u"Полина",
        "group" : u"БСТ1703",
        "age" : 20,
        "marks" : [5, 5, 5, 5]
    }
]

ave_rate = float(input("Введите среднюю оценку:  "))
print("\n")

def print_students(students):
    print( u"Имя студента".ljust(15), \
          u"Группа".ljust(8), \
          u"Возраст".ljust(8), \
          u"Оценки".ljust(20) )
    for student in students:
        print( student["name"].ljust(15), \
              student["group"].ljust(8), \
              str(student["age"]).ljust(8), \
              str(student["marks"]).ljust(20) )
        print("\n")

def average_rating(students, ave_rate):
    ave = []
    for student in students:
        if(float(sum(student["marks"]) / len(student["marks"])) >= ave_rate):
            ave.append(student["name"])

    for student in students:
        if(student["name"] in ave):
            print( student["name"].ljust(15), \
                student["group"].ljust(8), \
                str(student["age"]).ljust(8), \
                str(student["marks"]).ljust(20) )
            print("\n")


average_rating(groupmates, ave_rate)

File: blog/articles/static/groupmates.js
'use strict'

 let groupmates = [
    {
        "name" : "Гасан",
        "group" : "БСТ1703",
        "age" : 18,
        "marks" : [4, 4, 3, 5]
    },

    {
        "name" : "Марьяна",
        "group" : "БСТ1703",
        "age" : 20,
        "marks" : [2, 5, 4, 5]
    },

    {
        "name" : "Кирилл",
        "group" : "БСТ1703",
        "age" : 19,
        "marks": [3, 4, 4, 4]
    },

    {
        "name" : "Полина",
        "group" : "БСТ1702",
        "age" : 20,
        "marks" : [5, 5, 5, 5]
    },
    {
        "name" : "Александр",
        "group" : "БСТ1701",
        "age" : 20,
        "marks" : [2, 5, 4, 5]
    },
    {
        "name" : "Миша",
        "group" : "БСТ1702",
        "age" : 20,
        "marks" : [2, 5, 4, 5]
    }
]

let rpad = function(str,length){
    str=str.toString();
    while (str.length<length){
        str=str+' ';
    }
    return str;
}
let printStudents = function(students){
    console.log(
        rpad("Имя студента",15),
        rpad("Группа",8),
        rpad("Возраст",8),
        rpad("Оценки",20)
    );
    for(let i = 0; i<students.length;i++){
        console.log(
            rpad(students[i]['name'],15),
            rpad(students[i]['group'],8),
            rpad(students[i]['age'],8),
            rpad(students[i]['marks'],20),
            '\n'
        );
    }
};
printStudents(groupmates)


let filtredByGroup = function(students,groupNumber){
    for(let i =0;i<students.length;i++){
        if(groupNumber == students[i]['group']){
            console.log(
                rpad(students[i]['name'],15),
                rpad(students[i]['group'],8),
                rpad(students[i]['age'],8),
                rpad(students[i]['marks'],20),
                '\n'
            )
        }

    }
};


function makeWrap(){
    console.log("\n\n\n\n\n");
}

let filter = prompt('Введите номер группы');
makeWrap();
filtredByGroup(groupmates,filter);